File: scripts/calc_bounds.js

function tileToLatLng(x, y, zoom) {
    const n = Math.pow(2, zoom);
    const lon_deg = x / n * 360.0 - 180.0;
    const lat_rad = Math.atan(Math.sinh(Math.PI * (1 - 2 * y / n)));
    const lat_deg = lat_rad * (180.0 / Math.PI);
    return { lat: lat_deg, lng: lon_deg };
}

const zoom = 9;
// COVERAGE_BOUNDS: { 9: { xMin: 359, xMax: 443, yMin: 173, yMax: 229 } }
const x_min = 359;
const x_max = 443;
const y_min = 173;
const y_max = 229;

// Calculate corners
// Tile (x, y) covers from x to x+1, y to y+1
const nw = tileToLatLng(x_min, y_min, zoom); // Top-Left of first tile
const se = tileToLatLng(x_max + 1, y_max + 1, zoom); // Bottom-Right of last tile

console.log("West (Left): " + nw.lng.toFixed(4));
console.log("East (Right): " + se.lng.toFixed(4));
console.log("North (Top): " + nw.lat.toFixed(4));
console.log("South (Bottom): " + se.lat.toFixed(4));

File: scripts/calc_bounds.py
import math

def tile_to_latlng(x, y, zoom):
    n = 2.0 ** zoom
    lon_deg = x / n * 360.0 - 180.0
    lat_rad = math.atan(math.sinh(math.pi * (1 - 2 * y / n)))
    lat_deg = math.degrees(lat_rad)
    return lat_deg, lon_deg

zoom = 9
# COVERAGE_BOUNDS: { 9: { xMin: 359, xMax: 443, yMin: 173, yMax: 229 } }
# 注意：瓦片坐标通常是左上角。
# xMin (左) -> West
# xMax + 1 (右) -> East
# yMin (上) -> North
# yMax + 1 (下) -> South

x_min, x_max = 359, 443
y_min, y_max = 173, 229

north, west = tile_to_latlng(x_min, y_min, zoom)
south, east = tile_to_latlng(x_max + 1, y_max + 1, zoom)

print(f"Zoom Level: {zoom}")
print(f"X range: {x_min} - {x_max}")
print(f"Y range: {y_min} - {y_max}")
print("-" * 30)
print(f"West (Left): {west:.4f}")
print(f"East (Right): {east:.4f}")
print(f"North (Top): {north:.4f}")
print(f"South (Bottom): {south:.4f}")
